effects/zio/src/test/scala/sttp/client3/impl/zio/ZioTestBase.scala

```diff
@@ -1,35 +1,25 @@
 package sttp.client3.impl.zio
 
-import sttp.client3.testing.ConvertToFuture
-import zio.{Exit, Runtime, Task, Unsafe, durationInt}
+import scala.concurrent.Future
+
+import zio._
 
-import scala.concurrent.{Future, Promise}
-import scala.util.{Failure, Success}
+import sttp.client3.testing.ConvertToFuture
 
 trait ZioTestBase {
+
   private val runtime: Runtime[Any] = Runtime.default
 
   val convertZioTaskToFuture: ConvertToFuture[Task] = new ConvertToFuture[Task] {
     override def toFuture[T](value: Task[T]): Future[T] = {
-      val p = Promise[T]()
-
-      unsafeRunSync(value) match {
-        case Exit.Failure(c) =>
-          p.complete(
-            Failure(
-              c.failures.headOption.orElse(c.defects.headOption).getOrElse(new RuntimeException(s"Unknown cause: $c"))
-            )
-          )
-        case Exit.Success(v) => p.complete(Success(v))
+      Unsafe.unsafeCompat { implicit u =>
+        Runtime.default.unsafe.runToFuture(value.tapError { e =>
+          e.printStackTrace(); ZIO.unit
+        })
```

Comment:
I rewrote this toFuture conversion as the old one had execution issues in Scala-JS.

Reply:
runToFuture details here: https://github.com/zio/zio/blob/be586a263393af3962e9128d2336e1818f2ae6ef/core/shared/src/main/scala/zio/Runtime.scala#L159